static/js/message_edit.js

```diff
@@ -349,6 +347,10 @@ function edit_message(row, raw_content) {
 
     const show_edit_stream =
         message.is_stream && settings_data.user_can_move_messages_between_streams();
+    const is_editable =
+        editability === editability_types.TOPIC_ONLY ||
+        editability === editability_types.FULL ||
+        show_edit_stream;
```

Comment:
We can rename `show_edit_stream` to `is_stream_editable` here for more clarity.

Reply:
Yeah, that seems like a useful preparatory commit.

We also may want to rewrite `get_editability`'s interface to return an object that tells you which of content, topic, and stream are editable, potentially as a separate preparatory refactoring PR.  That function's interface has been really showing its age as we add new features like this :(.

Reply:
Renamed the variable.

```quote
We also may want to rewrite get_editability's interface to return an object that tells you which of content, topic, and stream are editable, potentially as a separate preparatory refactoring PR. That function's interface has been really showing its age as we add new features like this :(
```
Yes we should do this.